src/quantization/sq4_uniform_quantizer.h

```diff
@@ -254,11 +261,12 @@ void
 SQ4UniformQuantizer<metric>::ProcessQueryImpl(const DataType* query,
                                               Computer<SQ4UniformQuantizer>& computer) const {
     try {
-        computer.buf_ = new uint8_t[this->code_size_];  // todo: replace with allocator
+        computer.buf_ = reinterpret_cast<uint8_t*>(this->allocator_->Allocate(this->code_size_));
```

Comment:
We can move the memory mange logic to Computer class (allocate and deallocate) ?

like this:
```C++
void ProcessQueryImpl(...) {
    computer.AllocateBuffer(this->code_size_);
}

class Computer {
    // ...
    void Computer::AllocateBuffer(size_t size) {
        buf_ = allocator_->Allocate(size);
    }
    ~Computer() {
        if (buf_) {
            allocator_->Deallocate(buf_);
        }
    }
}
```

Reply:
the buf may not mean the memory buffer sometimes. it can point to an object such as a lookup table or other features to accumulate the distance compute. so it is difficult to manage the buf by computer.

Reply:
OK，here need guarantee the `Computer` and `Quantizer` instance destroy sequence